File: examples/key/decrypt_rsa/decrypt_rsa.go
package main

import (
	"fmt"
	"log"

	"github.com/bloock/bloock-sdk-go/v2/client"
	"github.com/bloock/bloock-sdk-go/v2/entity"
)

func main() {
	payload := "This will be encrypted"
	recordClient := client.NewRecordClient()
	encryptionClient := client.NewEncryptionClient()
	keyClient := client.NewKeyClient()

	key, err := keyClient.NewLocalKey(key.Rsa2048)
	if err != nil {
		log.Println(err)
	}

	// encryption ...

	fmt.Println("Trying to decrypt with the valid password")

	// To decrypt a record during the building process
	// we build a record from the encrypted record and add a decrypter
	decryptedRecord, err := recordClient.FromRecord(encryptedRecord).
		WithDecrypter(entity.NewRsaDecrypter(encryption.DecrypterArgs{
			LocalKey: &key,
		})).
		Build()
	if err != nil {
		log.Println(err)
	}

	// To decrypt an already encrypted record independently
	decryptedRecord, err = encryptionClient.Decrypt(encryptedRecord, entity.NewRsaDecrypter(encryption.DecrypterArgs{
		LocalKey: &key,
	}))
	if err != nil {
		log.Println(err)
	}

	fmt.Println("Decryption successful")
	fmt.Println("Decrypted payload: " + string(decryptedRecord.Retrieve()))
}

File: examples/key/decrypt_rsa/DecryptRsa.java
import com.bloock.sdk.client.EncryptionClient;
import com.bloock.sdk.client.KeyClient;
import com.bloock.sdk.client.RecordClient;
import com.bloock.sdk.entity.encryption.Encrypter;
import com.bloock.sdk.entity.key.KeyType;
import com.bloock.sdk.entity.key.LocalKey;
import com.bloock.sdk.entity.record.Record;

import java.nio.charset.StandardCharsets;

public class DecryptRsa {
        public static void main(String[] args) throws Exception {
                String payload = "This will be encrypted";

                // encryption ...
                Record encryptedRecord = null;

                RecordClient recordClient = new RecordClient();
                KeyClient keyClient = new KeyClient();

                LocalKey localKey = keyClient.newLocalKey(KeyType.Rsa2048);

                // To decrypt a record during the building process
                // we build a record from the encrypted record and add a decrypter
                Record decryptedRecord = recordClient.fromRecord(encryptedRecord)
                                .withDecrypter(new Encrypter(localKey))
                                .build();

                // To decrypt an already encrypted record independently
                EncryptionClient encryptionClient = new EncryptionClient();
                decryptedRecord = encryptionClient.decrypt(encryptedRecord, new Encrypter(localKey));

                System.out.println(
                                "Decrypted payload: " + new String(decryptedRecord.retrieve(), StandardCharsets.UTF_8));
        }
}